src/librustc/ty/layout.rs

```diff
@@ -1629,10 +1629,11 @@ impl<'a, 'tcx, C> TyLayoutMethods<'tcx, C> for Ty<'tcx>
 
             Variants::Single { index } => {
                 // Deny calling for_variant more than once for non-Single enums.
-                cx.layout_of(this.ty).map_same(|layout| {
-                    assert_eq!(layout.variants, Variants::Single { index });
-                    layout
-                });
+                <Result<TyLayout<'tcx>, !> as MaybeResult<Result<TyLayout<'tcx>, !>>>::from(
```

Comment:
@oli-obk This is the issue here:
```
error[E0308]: mismatched types==============================>        ] 118/137: rustc                                                                                       
    --> src/librustc/ty/layout.rs:1633:21
     |
1633 | /                     cx.layout_of(this.ty).to_result().map(|layout| {
1634 | |                         assert_eq!(layout.variants, Variants::Single { index });layout
1635 | |                     })
     | |______________________^ expected enum `std::result::Result`, found struct `rustc_target::abi::TyLayout`
     |
     = note: expected type `std::result::Result<std::result::Result<rustc_target::abi::TyLayout<'tcx, &'tcx ty::TyS<'tcx>>, !>, !>`
                found type `std::result::Result<rustc_target::abi::TyLayout<'_, &ty::TyS<'_>>, <<C as rustc_target::abi::LayoutOf>::TyLayout as ty::layout::MaybeResult<rustc_target::abi::TyLayout<'tcx, &'tcx ty::TyS<'tcx>>>>::Item>`
```

Not sure how to fix this. And I'm bugging you a lot for this issue :(